firedrake/slate/slate.py

```diff
@@ -60,8 +60,9 @@ class TensorBase(with_metaclass(ABCMeta)):
 
     def __init__(self):
         """Constructor for the TensorBase abstract class."""
+        # NOTE: This attribute is for caching kernels after
+        # an expression has been compiled.
         self._kernels = None
```

Comment:
Perhaps if you insert a blank line after this attribute that helps as a visual guide to the reader that this comment applies only to the immediately following line, and not for the any further lines in the same block. Alternatively, if you can get away with a shorter comment, then you could have it inline, in that case no blank lines needed as visual guides.

But probably it's even better if you just give this attribute a clearer name, e.g. `_kernel_cache` or even `_metakernel_cache`, maybe `_metakernels` or something similar.